csharp/src/Ice/CurrentExtensions.cs

```diff
@@ -0,0 +1,289 @@
+// Copyright (c) ZeroC, Inc.
+
+#nullable enable
+
+using Ice.Internal;
+using System.Diagnostics;
+
+namespace Ice;
+
+/// <summary>
+/// Provides extension methods for <see cref="Current"/>.
+/// </summary>
+public static class CurrentExtensions
+{
+    /// <summary>
+    /// Creates an outgoing response with reply status <see cref="ReplyStatus.Ok"/>.
+    /// </summary>
+    /// <typeparam name="TResult">The type of result.</typeparam>
+    /// <param name="current">The current object of the corresponding incoming request.</param>
+    /// <param name="result">The result to marshal into the response payload.</param>
+    /// <param name="marshal">The action that marshals result into an output stream.</param>
+    /// <param name="formatType">The class format.</param>
+    /// <returns>A new outgoing response.</returns>
+    public static OutgoingResponse createOutgoingResponse<TResult>(
+        this Current current,
+        TResult result,
+        Action<OutputStream, TResult> marshal,
+        FormatType formatType = FormatType.DefaultFormat)
+    {
+        OutputStream ostr = current.startReplyStream();
+        if (current.requestId != 0)
+        {
+            try
+            {
+                ostr.startEncapsulation(current.encoding, formatType);
+                marshal(ostr, result);
+                ostr.endEncapsulation();
+                return new OutgoingResponse(ostr);
+            }
+            catch (System.Exception exception)
+            {
+                return current.createOutgoingResponse(exception);
+            }
+        }
+        else
+        {
+            Debug.Fail("A one-way request cannot return a response");
+            return new OutgoingResponse(ostr);
+        }
+    }
+
+    /// <summary>
+    /// Creates an empty outgoing response with reply status <see cref="ReplyStatus.Ok"/>.
+    /// </summary>
+    /// <param name="current">The current object of the corresponding incoming request.</param>
+    /// <returns>An outgoing response with an empty payload.</returns>
+    public static OutgoingResponse createEmptyOutgoingResponse(this Current current)
+    {
+        OutputStream ostr = current.startReplyStream();
+        if (current.requestId != 0)
+        {
+            try
+            {
+                ostr.writeEmptyEncapsulation(current.encoding);
+            }
+            catch (System.Exception ex)
+            {
+                return current.createOutgoingResponse(ex);
+            }
+        }
```

Comment:
No debug fail here?

Reply:
No because it's ok (even expected) to create an empty response for a oneway request.

Reply:
Should probably be in all of these create response extension methods.